aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -437,6 +439,44 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     LOG.info("Successfully renamed table from {} to {}", from, to);
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
```

Comment:
I think we might want to have a feature flag like `glue.register-table.create-new-metadata-enabled` (for the lack of a better name) in `AwsProperties` to distinguish 2 behaviors, to either create a new metadata file or not. If the flag is false (by default), it can call the base class method directly.